ranger/ext/bind_arguments.py

```diff
@@ -0,0 +1,14 @@
+# This file is part of ranger, the console file manager.
+# License: GNU GPL version 3, see the file "AUTHORS" for details.
+
+from __future__ import (absolute_import, division, print_function)
+
+
+def bind_arguments(func, *part_args):
```

Comment:
Imo this is seriously overengineered and a lot *less* clear than `partial`, just do the wrapping manually in the few functions that actually need it, way more obvious and understandable.

Reply:
I agree that the solution is impracticable in the sense that `partial` already exist and there is no point in rewriting it. With that being said, I simply cannot see how writing something manually several times is better than using a design construct that is there and exists specifically to free us from the need of doing something manually.  I guess my problem here is on the principle and I am sorry but.... no. I cannot purposely write a less maintainable code that ignores a solution that out there and with that forces any other future developer to write the same thing manually, if they wish to extend the macro dict with a macro on their own.

Reply:
@Vifon, gonna let you decide on this. As you know I prefer FP. In this case however I don't think the unfamiliar style for the average python programmer is worth the effort saved over manually wrapping, what like 3 functions? I can make the changes if I have just need a decision.

@palikar, then the only bit that still needs your attention is the regex. We need to bring the two in line. Also though, to drive my point home about the brittleness of regexes. `[^%\s]*` doesn't work because it'll include parens in the macro for example. That's just a small oversight and an easy fix sure, but why not make it an impossible mistake to make? Though let's await @Vifon's opinion on that lazy dict thing too.

It's a little weird to me how you consider what `partial` does to be too menial to repeat manually but what the regex does (which is far more brittle and a common source of problems across languages even) to be perfectly fine duplicated in two places, in different ways that should be kept in sync with future changes even though there's no indication in either location that there even is something to be kept in sync.

Reply:
Oh, no.  I mean, if both regexes can be kept as one\at one place, I think that it should definitely be done. Albeit I am not sure if this is possible without some hackery. The way that the regex is stored in `_MacroTemplate` is in two separate pieces - a delimiter and the a thing that should be matched after the delimiter (or at least this is how I think string.Template work). In the macros extraction thing we need to search with the regex as a whole. I suppose something of the sort is possible

``` python
MACRO_REG = re.compile(_MacroTemplate.delimeter + _MacroTemplate.idpattern, re.IGNORECASE)
``` 
Or even probably better would be to keep all parts in a separate places and the construct the class (`_MacroTemplate`) and compile the regex with only what is needed. This would probably be the cleanest solution, if you agree to use the regex approach in extracting the macros. As I implied, `_MacroTemplate` will substitute only certain things that match the current regex. My idea - let's just use the same regex to extract the things that will potentially be substituted.